### ISSUE TITLE
Fix brew build

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#1324]
Seems the brew-macos test job has broken down after 0.15.1 release. Instead of cloning the repo from git it fetches the tar ball of the release and fails on the mismatching  checksum. Why did it start fetching release tar ball? Weird.

**Comment by plajjan** [comment]
This seems to have fixed itself after I merged a PR (despite the failing brew-macos test job, but it is not mandatory). I suppose the problem might be when main HEAD == the git tag for v0.15.1 or whatever version was most recently released? Then maybe brew handles this differently? We'll notice the next time we make a release I suppose. Even if so, I'm not sure it's worthwhile trying to fix. Homebrew is... difficult to work with. Closing issue.

**Comment by plajjan** [comment]
Nope, it's back. So it just intermittently went through *scratches head*. Very weird this stuff

**Comment by plajjan** [comment]
I really don't feel like spending time on this. I think Homebrew is quite terrible. The whole build process is a little too opinionated and essentially must run through multiple GitHub Action workflows. We need things we can run more locally to hack it up the way we want. I think this is the epitome of bad devops style.

The current process is instead to not continuously test the Homebrew Formula, rather we only use / test it once we make an actual release. In the past we had a lot of failures because all library dependencies of Acton were pulled in as Homebrew dependencies, for example, the Acton RTS uses libuv, so we pulled in libuv as a Homebrew dependency and then copied the static archive over into the Acton distribution. That meant we had lots of dependencies and as our stdlib was evolving, those dependencies would change. Now we have taken a different approach so all our dependencies are included as sources in the Acton distribution and we pull them down via git during the build. Thus the Homebrew Formula has gone from a bunch of dependencies to just a very small set of rather static dependencies. With this change the need to test the Homebrew Formula itself has also drastically been reduced. I don't feel we have a need to revive this but can rather let things remain as-is. The Homebrew test job was also rather slow, so from that perspective it's nice to skip it.

The latest issue we've had is that of GHC mismatches but I believe I've addressed this with #1627 so it really should start working quite reliably now.

Closing this issue.
